lib/utils/utf8.c

```diff
@@ -79,3 +80,37 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const char *str, size_t maxlen)
+{
+	int i = 0, nbyte = 0;
+	size_t len = strlen(str);
+	unsigned char *buf = (unsigned char *)str;
+
+	/* Return false when the string is greater than the maximum length. */
+	if (len > maxlen) {
+		return false;
+	}
+
+	while (i < len) {
+		if (buf[i] <= 0x7F && buf[i] >= 0x00) {
```

Comment:
```suggestion
		if (buf[i] <= ASCII_CHAR && buf[i] >= '\0') {
```

Reply:
I will encapsulate them, but I won't use explicit empty characters because it looks strange.

Reply:
>  I won't use explicit empty characters because it looks strange.
Do you mean the NULL terminator?

Reply:
I'll change it. Maybe it's not so offensive.